src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java

```diff
@@ -10,18 +11,27 @@
 public class DriverManager {
 
 	private Job2dDriver currentDriver = new LoggerDriver();
-
+	private CompositeDriver composite = new CompositeDriver();
```

Comment:
IMO: CompositeDriver should substitute currentDriver and help driver management but your approach is correct.

Reply:
We are not sure how to achieve it. We modified `DriverManager` a little bit in 18cd2d3 to return the composite as a current driver (which is now built earlier, not during the moment of returning it).

We had some ideas to check whether `driversToBeExecuted` contains (or not) a driver while adding or removing it. It partially worked – the logger wasn't added twice, but then we had an issue with checkbox in the additional drivers list that stayed checked in certain circumstances (when logger was disabled – i.e. changed to a different driver – from the main Drivers list).

One idea is to somehow update the status of the checkbox if a driver is "additional", but we currently decided not to dig deeper into it tonight since it's already very late.

Would you share any suggestions with us on that, please? (However can't promise we will take up the gauntlet since we still many many other tasks to complete :))

Reply:
It was only my opinion about the possibility to move more responsibilities to the composite, but your solution is valid.